### DISCUSSION TITLE
Socks5 Not Work on Remote Device

### DISCUSSION THREAD

**Discussion by bijanbina** [rustdesk/rustdesk#6398] [General]
### Bug Description
When you set socks5 proxy on Device A, Device A uses socks5 to connect to any remote machine
The issue is when other devices connect to Device A connection is not routed through socks5 

In case this is too much work can you point me to the right point in the source code?

### How to Reproduce
Set socks5 on Dev A, try to connect to it through Dev B

Windows 10,  RustDesk 1.2.1

**Comment by pdifeo** [comment]
Same problem on ver 1.2.3. 

To verify that the proxy socks work, I configured Firefox and everything is fine. Rustdesk continues to report "Connecting". In the firewall logs there is no trace of the connections.

**Comment by bijanbina** [comment]
I think issue lies in port hole punching and that stuff fails but its too complicated for me to pickup

**Comment by hang2ran** [comment]
This bug is very old, has it not been solved yet?

**Reply by bijanbina** [reply]
no

**Reply by hang2ran** [reply]
I have tested from version 1.1.9 to the latest version, but I cannot use socks5 to connect online normally. It always shows that it is connecting to the network.

**Reply by hr0109** [reply]
what i found
1、It seems that RustDesk can only recognize 127.0.0.1:xxxx，anything other than this will not work
2、RustDesk use udp，so make sure your socks5 server can deal with udp correctly

**Reply by jm789** [reply]
He @hr0109 from your comment above it sounds like you've been able to get a socks5 proxy working? Please can you point me in the right direction?

**Comment by jm789** [comment]
Adding some info / questions here - seems to be related.

n order to get past firewalls, I'm trying to get the Rustdesk client to talk to the Rustdesk Self Hosted Server (Non-Pro version) via a proxy. Tried many proxies over the past few weeks, but none seem to work. Tinyproxy, dante, squid, nhinx, socks5 etc.

Trying to get this going:

client <-------> proxy.example.com:1080 <----------> rustdesk.example.com:21115-21119

- Client is running on MacOS.
- Proxy is a socks5 proxy with TCP and UDP ports open on 1080. Proxy handles tcp and udp.
- rustdesk.example.com is the non-pro version. 
- Without proxy, client can connect successfully to server and rustdesk operation works as expected.
- Safari on client MacOs can successfully pass all traffic through proxy, no problem.

When doing a Wireshark on the rustdesk/proxy, I see that following:

1. Rustdesk Client sends socks request announcing no authentication to proxy.
2. Proxy replies with Accept
3. Client sends socks request to proxy, containing in the payload a connect request for rustdesk.example.com:21114
4.  Proxy forwards request to rustdesk.example.com:21114
5.  rustdesk.example.com rejects - port 21114 is not even active
6. Proxy replies to client with reject.
7. Process starts over - Client seems to be stuck here, instead of trying to connect to ID or Relay server via proxy.

What is supposed to happen?
Why is the client sending connect request to the server on port 21114? Can this be disabled /avoided?
